packaging_automation/common_tool_methods.py

```diff
@@ -290,3 +330,151 @@ def remove_cloned_code(exec_path: str):
+        Jinja2 is used as template engine and render function gets the file change parameters in the file with the given
+        parameters as input and returns the output '''
+    minor_version = get_minor_project_version(project_version)
+    env = get_template_environment(templates_path)
+    template = env.get_template(template_file_path)
+    return f"{template.render(project_version=project_version, project_minor_version=minor_version)}\n"
+
+
+def write_to_file(content: str, dest_file_name: str):
+    with open(dest_file_name, "w") as writer:
+        writer.write(content)
+
+
+def get_gpg_fingerprints_by_name(name: str) -> List[str]:
+    '''Returns GPG fingerprint by its unique key name. We use this function to determine the fingerprint that
+       we should use when signing packages'''
+    result = subprocess.run(shlex.split(f"gpg --list-keys "), check=True, stdout=subprocess.PIPE)
+    lines = result.stdout.decode("ascii").splitlines()
+    finger_prints = []
+    previous_line = ""
+    for line in lines:
+        if line.startswith("uid") and name in line:
+            finger_prints.append(previous_line.strip())
+            continue
+        previous_line = line
+    return finger_prints
+
+
+def delete_gpg_key_by_name(name: str, key_type: GpgKeyType):
+    keys = get_gpg_fingerprints_by_name(name)
+
+    # There could be more than one key with the same name. While statement is used to delete all the public keys
+    # until no key remains (i.e. key_id is empty).
+    # Public and private keys are stored with the same fingerprint. In some cases one of them may not be exist.
+    # Therefore non-existence case is possible
+    for key_id in keys:
+        delete_command = f"gpg --batch --yes --delete-key {key_id}" if key_type == GpgKeyType.public \
+            else f"gpg --batch --yes --delete-secret-key {key_id}"
+        output = run_with_output(delete_command)
+        if output.returncode == 0:
+            print(f"{key_type.name.capitalize()} key with the id {key_id} deleted")
+        #
+        elif output.returncode == 2:  # Key does not exist in keyring
+            continue
+        else:
+            print(f"Error {output.stderr.decode('ascii')}")
+            break
+
+
+def delete_public_gpg_key_by_name(name: str):
+    delete_gpg_key_by_name(name, GpgKeyType.public)
+
+
+def delete_private_gpg_key_by_name(name: str):
+    delete_gpg_key_by_name(name, GpgKeyType.private)
+
+
+def delete_all_gpg_keys_by_name(name: str):
+    delete_private_gpg_key_by_name(name)
+    delete_public_gpg_key_by_name(name)
+
+
+def get_secret_key_by_fingerprint_without_password(fingerprint: str) -> str:
+    # gpg.export_keys needs to get password. Otherwise it gives error. Dummy password is only to bypass parameter
+    # error
+    dummy_password = "123"
+
+    return get_secret_key_by_fingerprint_with_password(fingerprint, dummy_password)
```

Comment:
could you check gpg manuel to see if there is a better way to do that, instead of passing a dummy password ?

Reply:
Fixed by upgrading the library